File: objectModel/CSharp/Microsoft.CommonDataModel.ObjectModel/Persistence/CdmFolder/Projections/OperationAddAttributeGroupPersistence.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

namespace Microsoft.CommonDataModel.ObjectModel.Persistence.CdmFolder
{
    using Microsoft.CommonDataModel.ObjectModel.Cdm;
    using Microsoft.CommonDataModel.ObjectModel.Enums;
    using Microsoft.CommonDataModel.ObjectModel.Persistence.CdmFolder.Types;
    using Microsoft.CommonDataModel.ObjectModel.Utilities;
    using Newtonsoft.Json.Linq;

    /// <summary>
    /// Operation AddAttributeGroup persistence
    /// </summary>
    public class OperationAddAttributeGroupPersistence
    {
        public static CdmOperationAddAttributeGroup FromData(CdmCorpusContext ctx, JToken obj)
        {
            if (obj == null)
            {
                return null;
            }

            CdmOperationAddAttributeGroup addAttributeGroupOp = OperationBasePersistence.FromData<CdmOperationAddAttributeGroup>(ctx, CdmObjectType.OperationAddAttributeGroupDef, obj);
            addAttributeGroupOp.AttributeGroupName = obj["attributeGroupName"]?.ToString();

            return addAttributeGroupOp;
        }

        public static OperationAddAttributeGroup ToData(CdmOperationAddAttributeGroup instance, ResolveOptions resOpt, CopyOptions options)
        {
            if (instance == null)
            {
                return null;
            }

            OperationAddAttributeGroup obj = OperationBasePersistence.ToData<OperationAddAttributeGroup>(instance, resOpt, options);
            obj.AttributeGroupName = instance.AttributeGroupName;

            return obj;
        }
    }
}

File: objectModel/Java/objectmodel/src/main/java/com/microsoft/commondatamodel/objectmodel/persistence/cdmfolder/projections/OperationAddAttributeGroupPersistence.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.persistence.cdmfolder.projections;

import com.fasterxml.jackson.databind.JsonNode;
import com.microsoft.commondatamodel.objectmodel.cdm.CdmCorpusContext;
import com.microsoft.commondatamodel.objectmodel.cdm.projections.CdmOperationAddAttributeGroup;
import com.microsoft.commondatamodel.objectmodel.enums.CdmObjectType;
import com.microsoft.commondatamodel.objectmodel.persistence.cdmfolder.types.projections.OperationAddAttributeGroup;
import com.microsoft.commondatamodel.objectmodel.utilities.CopyOptions;
import com.microsoft.commondatamodel.objectmodel.utilities.ResolveOptions;

/**
 * Operation AddAttributeGroup persistence
 */
public class OperationAddAttributeGroupPersistence {
    public static CdmOperationAddAttributeGroup fromData(final CdmCorpusContext ctx, final JsonNode obj) {
        if (obj == null) {
            return null;
        }

        CdmOperationAddAttributeGroup addAttributeGroupOp = OperationBasePersistence.fromData(ctx, CdmObjectType.OperationAddAttributeGroupDef, obj);

        if (obj.get("attributeGroupName") != null) {
            addAttributeGroupOp.setAttributeGroupName(obj.get("attributeGroupName").asText());
        }

        return addAttributeGroupOp;
    }

    public static OperationAddAttributeGroup toData(final CdmOperationAddAttributeGroup instance, final ResolveOptions resOpt, final CopyOptions options) {
        if (instance == null) {
            return null;
        }

        OperationAddAttributeGroup obj = OperationBasePersistence.toData(instance, resOpt, options);
        obj.setAttributeGroupName(instance.getAttributeGroupName());

        return obj;
    }
}
